lib/node_modules/@stdlib/lapack/blas/dlaswp/lib/dlaswp.js

```diff
@@ -0,0 +1,151 @@
+*/
+
+'use strict';
+
+// MODULES //
+
+var isLayout = require( '@stdlib/blas/base/assert/is-layout' );
+var floor = require( '@stdlib/math/base/special/floor' );
+var dswap = require( '@stdlib/blas/base/dswap' );
+
+
+// MAIN //
+
+/**
+* Performs a series of row interchanges on the matrix `A` using pivot indices stored in `IPIV`.
+*
+* @param {string} order - storage layout
+* @param {PositiveInteger} N - number of columns of `A`
+* @param {Float64Array} A - matrix
+* @param {PositiveInteger} LDA - leading dimension of `A`
+* @param {PositiveInteger} k1 - index of first row to interchange
+* @param {PositiveInteger} k2 - index of last row to interchange
+* @param {Int32Array} IPIV - vector of pivot indices
+* @param {PositiveInteger} incX - increment between successive values of `IPIV`
+* @returns {Float64Array} permuted matrix `A`
+*
+* @example
+* var Int32Array = require( '@stdlib/array/int32' );
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var IPIV = new Int32Array( [ 2, 0, 1 ] );
+* var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+*
+* dlaswp( 'column-major', 3, A, 2, 0, 2, IPIV, 1 );
+* // A => <Float64Array>[ 2.0, 3.0, 4.0, 5.0, 1.0, 6.0 ]
+*/
+function dlaswp( order, N, A, LDA, k1, k2, IPIV, incX ) {
+	var IPIV;
+	var incX;
+	var inc;
+	var ix0;
+	var LDA;
+	var n32;
+	var tmp;
+	var i2;
+	var ip;
+	var ix;
+	var k1;
+	var k2;
+	var i1;
+	var A;
+	var i;
+	var j;
+	var k;
+	var N;
+
+	if ( !isLayout( order ) ) {
+		throw new TypeError( 'invalid argument. First argument must be a valid order. Value: `%s`.', order );
+	}
+
+	if ( incX > 0 ) {
+		ix0 = k1;
+		i1 = k1;
+		i2 = k2;
+		inc = 1;
+	} else if ( incX < 0 ) {
+		ix0 = k1 + ( k1 - k2 ) * incX;
+		i1 = k2;
+		i2 = k1;
+		inc = -1;
+	} else {
+		return A;
+	}
+
+	if ( order === 'row-major' ) {
+		ix = ix0;
+		if ( inc === 1 ) {
+			for ( i = i1; i <= i2; i++ ) {
+				ip = IPIV[ ix ];
+				if ( ip !== i ) {
+					dswap( N, A, i * LDA, A, ip * LDA);
+				}
+				ix += incX;
+			}
+			return A;
+		}
+		for ( i = i1; i >= i2; i-- ) {
+			ip = IPIV[ ix ];
+			if ( ip !== i ) {
+				dswap( N, A, i * LDA, A, ip * LDA);
+			}
+			ix += incX;
+		}
+		return A;
+	} else {
+		n32 = floor( N / 32 ) * 32;
+		if ( n32 !== 0 ) {
+			for ( j = 1; j < n32; j += 32 ) {
+				ix = ix0;
+				for ( i = i1; i < i2; i += inc ) {
+					ip = IPIV[ ix ];
+					if ( ip !== i ) {
+						for ( k = j; k < j + 31; k++ ) {
+							tmp = A[ ( k - 1 ) * LDA + i ];
+							A[ ( k - 1 ) * LDA + i ] = A[ ( k - 1 ) * LDA + ip ];
+							A[ ( k - 1 ) * LDA + ip ] = tmp;
+						}
+					}
+					ix += incX;
+				}
+			}
+		}
+		if ( n32 !== N ) {
+			n32 += 1;
+			ix = ix0;
+			for ( i = i1; i < i2; i += inc ) {
+				ip = IPIV[ ix ];
+				if ( ip !== i ) {
+					for ( k = n32; k < N; k++ ) {
+						tmp = A[ ( k - 1 ) * LDA + i ];
+						A[ ( k - 1 ) * LDA + i ] = A[ ( k - 1 ) * LDA + ip ];
+						A[ ( k - 1 ) * LDA + ip ] = tmp;
+					}
+				}
+				ix += incX;
+			}
+		}
+		return A;
+	}
+}
```

Comment:
I am willing to get a review on implementation. I'll soon address comments related to linting, variable declaration convention, and description.

Reply:
`gonum` implementation restricts for `incX=1`, https://github.com/gonum/gonum/blob/master/lapack/gonum/dlaswp.go#L48

Reply:
https://github.com/gonum/gonum/blob/master/lapack/gonum/dlaswp.go#L34

Reply:
I am not sure why they have that restriction, and I don't think we need to follow suit.

Reply:
Yeah, there really isn't a reason for us to only accommodate `incX = +-1`. I should state this explicitly: `gonum` is a source of inspiration; it is **not** a reference implementation.

Reply:
Agreed, I modified implementation myself. I think it is fine, will wait for your review on it.

Reply:
Seems okay.